codes/go/chapter_array_and_linkedlist/array_test.go

```diff
@@ -0,0 +1,145 @@
+package chapter_array_and_linkedlist
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestInitArray(t *testing.T) {
+	t.Parallel()
+	var arr [5]int
+	for _, v := range arr {
+		if v != 0 {
+			t.Fatalf("array init exception")
+		}
+	}
+}
+
+func TestRandomAccess(t *testing.T) {
+	t.Parallel()
+	min, max := 1, 5
+	nums := [5]int{min, 2, 3, 4, max}
+	ans := randomAccess(nums)
+	if ans < min || ans > max {
+		t.Fatalf("Expected range is greater than min: %v and less than max: %v, got ans: %v", min, max, ans)
+	}
+}
+
+func TestExtend(t *testing.T) {
+	t.Parallel()
+	nums := [5]int{1, 2, 3, 4, 5}
+	newNums := extend(nums)
```

Comment:
Can we add an argument of `enlarge` to this function `extend(nums, enlarge)` to keep the consistency with other languages?

Reply:
Please see my comments below https://github.com/krahets/hello-algo/pull/178#issuecomment-1365704881